packages/flutter/lib/src/material/ink_well.dart

```diff
@@ -616,6 +616,13 @@ class _InkResponseState<T extends InkResponse> extends State<T> with AutomaticKe
 /// ancestor to the ink well). The [MaterialType.transparency] material
 /// kind can be used for this purpose.
 ///
+/// ### The ink splashes are unpleasantly animating with transitioning widgets!
```

Comment:
I think this could be a little clearer.

The original issue made the problem pretty clear: [InkWell doesn't work well in AnimatedContainer](https://github.com/flutter/flutter/issues/5211).

Maybe this section could be called: InkWell's splashes don't track the size of an animated container

Start with paragraph that explains the symptoms of the problem (we want to the reader to be able to quickly confirm that this section is about the problem they're seeing). You could even include the example from the original issue.

This next paragraph is correct but probably a bit difficult to understand "transitions should not be placed inside Material" sounds draconian and we haven't tied Material and InkWell together here.  The overall point is that an InkWell's splashes will not change to conform to changes in the size of its underlying Material - where the splashes are actually rendered.  Typically the underlying Material comes from the MaterialApp and is not animated. We're saying that app's shouldn't create InkWells within Material widgets whose size is also animated. 

Once that's clear you could reiterate the bit about LayoutChangedNotification. However I don't think that will help most readers.

Reply:
Sounds good. I redid the explanation and title and decided to remove the LayoutChangedNotification paragraph, since it will likely not be helpful for our readers.

Reply:
After investigation, decided on an explanation as to why the ink effects to appear unusually. I also added a sample app to demonstrate the behavior.